src/runtime_src/core/tools/xbutil2/OO_AieClockFreq.cpp

```diff
@@ -0,0 +1,187 @@
+/**
+* Copyright (C) 2022 Xilinx, Inc
+*
+* Licensed under the Apache License, Version 2.0 (the "License"). You may
+* not use this file except in compliance with the License. A copy of the
+* License is located at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations
+* under the License.
+*/
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "core/common/device.h"
+#include "core/common/query_requests.h"
+#include "OO_AieClockFreq.h"
+#include "tools/common/XBUtilities.h"
+
+namespace XBU = XBUtilities;
+
+// 3rd Party Library - Include Files
+#include <boost/algorithm/string/join.hpp>
+#include <boost/format.hpp>
+#include <boost/program_options.hpp>
+
+namespace po = boost::program_options;
+namespace qr = xrt_core::query;
+
+// System - Include Files
+#include <iostream>
+
+#define to_mega(x) x/1000000
+
+// ----- H E L P E R M E T H O D S ------------------------------------------
+static double get_aie_part_freq(const std::shared_ptr<xrt_core::device>& device, uint32_t part_id)
+{
+  double freq = 0;
+  try {
+    freq = static_cast<double>(to_mega(xrt_core::device_query<qr::aie_get_freq>(device, part_id)));
+  }
+  catch (const std::exception &e) {
+    std::cerr << boost::format("ERROR: Failed to read clock frequency of AIE partition(%d)\n %s\n") % part_id % e.what();
+    throw xrt_core::error(std::errc::operation_canceled);
+  }
+
+  return freq;
+}
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+OO_AieClockFreq::OO_AieClockFreq( const std::string &_longName, bool _isHidden )
+    : OptionOptions(_longName, _isHidden, "AIE clock frequency operations" )
+    , m_device("")
+    , m_partition_id(1)
+    , m_get(false)
+    , m_freq("")
```

Comment:
```suggestion
    , m_setFreq("")
```
It just makes the code easier to read for the next person.